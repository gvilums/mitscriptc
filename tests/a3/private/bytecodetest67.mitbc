function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [5],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_const	0
				return
			]
		},
		function
		{
			functions = [],
			constants = ["hey"],
			parameter_count = 1,
			local_vars = [x],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_local	0
				load_const	0
				add
				return
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [x],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_local	0
				return
			]
		}
	],
	constants = [4, 1, "sup", "wo", None, "they equal", 5],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [thingsILike, str, cat, eating, doinghw, drawing, wo, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	7
		load_func	1
		alloc_closure	0
		store_global	8
		load_func	2
		alloc_closure	0
		store_global	9
		alloc_record
		dup
		dup
		dup
		dup
		load_const	0
		field_store	3
		load_const	1
		neg
		field_store	4
		load_const	0
		field_store	5
		load_func	3
		alloc_closure	0
		field_store	6
		store_global	0
		load_global	7
		load_global	0
		field_load	6
		call	0
		call	1
		pop
		load_global	0
		load_func	4
		alloc_closure	0
		field_store	6
		load_global	7
		load_global	0
		field_load	6
		load_const	2
		call	1
		call	1
		pop
		load_const	3
		store_global	1
		load_global	0
		load_global	1
		load_func	5
		alloc_closure	0
		index_store
		load_global	7
		load_global	0
		field_load	6
		load_const	4
		call	1
		call	1
		pop
		load_global	7
		load_global	0
		load_global	1
		index_load
		load_const	4
		call	1
		call	1
		pop
		load_global	7
		load_global	0
		load_global	1
		index_load
		load_global	0
		field_load	6
		load_const	4
		call	1
		call	1
		call	1
		pop
		load_global	0
		alloc_record
		field_store	6
		load_global	7
		load_global	0
		field_load	6
		call	1
		pop
		load_global	0
		field_load	6
		store_global	2
		load_global	2
		load_global	0
		field_load	6
		eq
		not
		if	6
		load_global	7
		load_const	5
		call	1
		pop
		goto	1
		load_global	0
		load_const	6
		field_store	6
		load_global	7
		load_global	0
		field_load	6
		call	1
		pop
	]
}