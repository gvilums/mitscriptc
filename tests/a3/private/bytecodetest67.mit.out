5
suphey
None
None
None
{}
they equal
5
